sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs

```diff
@@ -222,9 +222,13 @@ private static IDataTypeInfo GetDataTypeInfo(Type propertyType, IMemberNameConve
             static bool IsNullableType(Type type) =>
                 type.IsConstructedGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>);
 
-            if (s_primitiveTypeMap.TryGetValue(propertyType, out SearchFieldDataType SearchFieldDataType))
+            if (s_primitiveTypeMap.TryGetValue(propertyType, out SearchFieldDataType searchFieldDataType))
             {
-                return DataTypeInfo.Simple(SearchFieldDataType);
+                return DataTypeInfo.Simple(searchFieldDataType);
+            }
+            else if (SpatialProxyFactory.CanCreate(propertyType))
+            {
+                return DataTypeInfo.Simple(SearchFieldDataType.GeographyPoint);
```

Comment:
Is that saying turn ANY spatial type into a point?  Don't we support other things besides points that can be used for queries but not index fields?

Reply:
No, only `GeographyPoint`.

Reply:
But doesn't SpatialProxyFactory support Points, Polygons, and Lines?  If somebody has a property of type `Microsoft.Spatial.GeographyPolygon`, doesn't this convert it into a point in the index?

Reply:
Good catch. You're right. This specific code was from an earlier point before I added more support to `SpatialProxyFactory` and forgot to change it, apparently.

Reply:
New tests have been added as well. Clearly was missing some needed coverage here.